src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -131,3 +139,108 @@ export const mapTokenUi = ({
 		$exchanges
 	})
 });
+
+/**
+ * Type guard to check if a token is of type RequiredTokenWithLinkedData.
+ * This checks whether the token has a twinTokenSymbol field and ensures that it is a string.
+ *
+ * @param token - The token object to be checked.
+ * @returns A boolean indicating whether the token is of type RequiredTokenWithLinkedData.
+ */
+export const isRequiredTokenWithLinkedData = (token: Token): token is RequiredTokenWithLinkedData =>
+	'twinTokenSymbol' in token && typeof token.twinTokenSymbol === 'string';
+
+/**
+ * Type guard to check if an object is of type TokenGroupUi.
+ *
+ * @param obj - The object to check.
+ * @returns A boolean indicating whether the object is a TokenGroupUi.
+ */
+export const isTokenGroupUi = (obj: unknown): obj is TokenGroupUi =>
```

Comment:
I am not really sure I understand what is the use case here: why the input is `unkown` type? where is this one expected to be used? can we not limit the possible inputs?

Reply:
01f16459